library/src/main/java/com/climate/mirage/tasks/BitmapTask.java

```diff
@@ -119,14 +147,54 @@ public Bitmap doTask(Void... params) throws MirageIOException {
 						putResultInDiskCache(bitmap);
 					}
 				}
-				// SOURCE was taken care of on loading.
+				// saving the source is taken care of during loading
+                // since it must directly write to the file
 			}
 		}
 
 		return !isCancelled() ? bitmap : null;
 	}
 
-	abstract protected Bitmap loadFromExternal() throws IOException;
+    private int determineSampleSize(BitmapFactory.Options outOpts) {
+        int dimen = Math.max(outOpts.outWidth, outOpts.outHeight);
+        float div = dimen / (float)request.getResizeTargetDimen();
+		if (div < 1) return 1;
```

Comment:
When we first started mirage we were using tabs rather than spaces. Now I'm using spaces. I'll do a complete reformat (pass as a separate PR) with hopefully a "ignore white space" diff of nothing.